Unstable Operator test KeycloakDeploymentTest.testKeycloakDeploymentBeforeSecret

### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the [latest nightly release](https://github.com/keycloak/keycloak/releases/tag/nightly)

### Area

operator

### Describe the bug

When running https://github.com/keycloak/keycloak/actions/runs/6724183110/job/18276241363?pr=23343, I got the error message for an unstable test: 

```
[ERROR] org.keycloak.operator.testsuite.integration.KeycloakDeploymentTest.testKeycloakDeploymentBeforeSecret  Time elapsed: 1.118 s  <<< ERROR!
java.lang.NullPointerException: Cannot invoke "org.keycloak.operator.crds.v2alpha1.deployment.KeycloakStatus.findCondition(String)" because "kcStatus" is null
	at org.keycloak.operator.testsuite.utils.CRAssert.assertKeycloakStatusCondition(CRAssert.java:56)
	at org.keycloak.operator.testsuite.utils.CRAssert.assertKeycloakStatusCondition(CRAssert.java:44)
	at org.keycloak.operator.testsuite.utils.CRAssert.assertKeycloakStatusCondition(CRAssert.java:39)
	at org.keycloak.operator.testsuite.integration.KeycloakDeploymentTest.lambda$testKeycloakDeploymentBeforeSecret$1(KeycloakDeploymentTest.java:113)
	at org.awaitility.core.AssertionCondition.lambda$new$0(AssertionCondition.java:53)
	at org.awaitility.core.ConditionAwaiter$ConditionPoller.call(ConditionAwaiter.java:232)
	at org.awaitility.core.ConditionAwaiter$ConditionPoller.call(ConditionAwaiter.java:219)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

### Version

main

### Expected behavior

Test shouldn't be flaky

### Actual behavior

Test if flaky (previous run was green, and I only changed some docs)

### How to Reproduce?

?

### Anything else?

_No response_

--- a/operator/src/test/java/org/keycloak/operator/testsuite/integration/KeycloakDeploymentTest.java
+++ b/operator/src/test/java/org/keycloak/operator/testsuite/integration/KeycloakDeploymentTest.java
@@ -107,7 +107,7 @@ public class KeycloakDeploymentTest extends BaseOperatorTest {
         Resource<StatefulSet> stsResource = k8sclient.resources(StatefulSet.class).withName(deploymentName);
         Resource<Keycloak> keycloakResource = k8sclient.resources(Keycloak.class).withName(deploymentName);
         // expect no errors and not ready, which means we'll keep reconciling
-        Awaitility.await().untilAsserted(() -> {
+        Awaitility.await().ignoreExceptions().untilAsserted(() -> {
             assertThat(stsResource.get()).isNotNull();
             Keycloak keycloak = keycloakResource.get();
             CRAssert.assertKeycloakStatusCondition(keycloak, KeycloakStatusCondition.HAS_ERRORS, false);

